2.17.0 (#478)
percona/pmm-doc: docs/release-notes/2.17.0.md

    </tr>
    <tr class="field-even field">
      <th class="field-name">Installation:</th>
      <td class="field-body">
        <a class="reference external" href="https://www.percona.com/software/pmm/quickstart">Installing Percona Monitoring and Management</a></td>
    </tr>
  </tbody>
</table>

Percona Monitoring and Management (PMM) is a free and open-source platform for managing and monitoring MySQL, MongoDB, and PostgreSQL performance.

## Release Highlights

- Custom Certificates support for MySQL monitoring, including remote monitoring, will let you monitor MySQL instances with such security restrictions. This functionality might help monitor Google Cloud SQL for MySQL instances easier.

- External monitoring cleanups on UI to make it easier to use. You can fill all parameters or provide the URL of your endpoint, and PMM will detect all needed parts.

- pg_stat_monitor 0.9.0 support. This change will give you compatibility with the latest version. More features coming later.

- Single-line install of PMM Server on supported Linux distros (this feature is in Technical Preview).

- DBaaS Changes: (this feature is in Technical Preview)

>     - DBaaS is now is better open for the public, so you can quickly turn it ON  on the Advanced settings page. ([Read more](../how-to/configure.md#advanced-settings))

Reviewer: ```suggestion
    - Make it easier to preview DBaaS functionality, so users can quickly turn it ON  on the Advanced settings page. ([Read more](../how-to/configure.md#advanced-settings))
```

    - Database components management will enable PMM administrators to restrict some DB versions from creating DB Clusters.
    - For PXC clusters created using DBaaS, the HAProxy will be used by default.  Please note: the proper monitoring HAProxy with DBaaS will come later.

- Changes to *Sign in to Percona Platform*. From this release, Registration of the Percona account will be more secure and require additional confirmation.

## New Features

> * [PMM-7863](https://jira.percona.com/browse/PMM-7863): Add version management for HAProxy

Reviewer: ```suggestion
* [PMM-7863](https://jira.percona.com/browse/PMM-7863): DBaaS: Ability to specify in K8s configuration the version of HAProxy to be used for DB creation 
```

* [PMM-7848](https://jira.percona.com/browse/PMM-7848), [PMM-7847](https://jira.percona.com/browse/PMM-7847), [PMM-7421](https://jira.percona.com/browse/PMM-7421): Add support for using SSL certificates between pmm-admin and monitored MySQL databases
* [PMM-7883](https://jira.percona.com/browse/PMM-7883): Single-line install of PMM Server on supported Linux distros - [Technical Preview]
* [PMM-7013](https://jira.percona.com/browse/PMM-7013), [PMM-7819](https://jira.percona.com/browse/PMM-7819): Use HAProxy by default instead of ProxySQL

> * [PMM-7356](https://jira.percona.com/browse/PMM-7356), [PMM-7581](https://jira.percona.com/browse/PMM-7581): Management of available versions of components

Reviewer: ```suggestion
* [PMM-7356](https://jira.percona.com/browse/PMM-7356), [PMM-7581](https://jira.percona.com/browse/PMM-7581): DBaaS: Management of available and recommended versions of DB components
```

> * [PMM-7358](https://jira.percona.com/browse/PMM-7358), [PMM-7576](https://jira.percona.com/browse/PMM-7576): Default version for Create DB cluster

Reviewer: ```suggestion
* [PMM-7358](https://jira.percona.com/browse/PMM-7358), [PMM-7576](https://jira.percona.com/browse/PMM-7576): DBaaS: Management of default versions of DB components

```

* [PMM-7064](https://jira.percona.com/browse/PMM-7064): Integrated Alerting: Presenting severity of the Alert Rule using different colors
* [PMM-7946](https://jira.percona.com/browse/PMM-7946): Better error message on PMM client if server doesn't support HAProxy
* [PMM-7932](https://jira.percona.com/browse/PMM-7932): Usability improvements on UI for adding External Services
* [PMM-7641](https://jira.percona.com/browse/PMM-7641), [PMM-7820](https://jira.percona.com/browse/PMM-7820): Make DBAAS feature flag - same as other Feature Flags
* [PMM-7966](https://jira.percona.com/browse/PMM-7966): Telemetry: Collect enabled/disabled status for Integrated Alerting and Security Threat Tool features

## Bugs Fixed

* [PMM-7911](https://jira.percona.com/browse/PMM-7911): DBaaS: Invalid Number of Nodes results in bad request
* [PMM-7884](https://jira.percona.com/browse/PMM-7884): DBaaS: Fix DB Cluster tab loading
* [PMM-7917](https://jira.percona.com/browse/PMM-7917): Postgres exporter has high CPU usage during Restart
* [PMM-8037](https://jira.percona.com/browse/PMM-8037): Platform Sign in flow on PMM broken

> * [PMM-7702](https://jira.percona.com/browse/PMM-7702): DBaaS: Cannot edit PSMDB clusters

Reviewer: ```suggestion
* [PMM-7702](https://jira.percona.com/browse/PMM-7702): DBaaS: Cannot edit already created PSMDB clusters
```

* [PMM-7991](https://jira.percona.com/browse/PMM-7991): MySQL Summary panel doesn't exist on MySQL Summary dashboard

> * [PMM-7939](https://jira.percona.com/browse/PMM-7939): Inconsistent reported version

Reviewer: ```suggestion
* [PMM-7939](https://jira.percona.com/browse/PMM-7939): Inconsistent format of version reporting in pmm-admin 
```

* [PMM-7920](https://jira.percona.com/browse/PMM-7920): Postgres Exporter has increased memory usage with pmm-client 2.15.1 & pmm-server 2.16.0

> * [PMM-7700](https://jira.percona.com/browse/PMM-7700): Rule API crashing - undefined params

Reviewer: ```suggestion
* [PMM-7700](https://jira.percona.com/browse/PMM-7700): Integrated Alerting:  Rule API crashing with more than two parameters or invalid values 
```

# Percona Monitoring and Management 2.17.0

<table class="docutils field-list" frame="void" rules="none">
  <colgroup>
    <col class="field-name">
    <col class="field-body">
  </colgroup>
  <tbody valign="top">
    <tr class="field-odd field">
      <th class="field-name">Date:</th>
      <td class="field-body">May 10, 2021</td>
    </tr>
    <tr class="field-even field">
      <th class="field-name">Installation:</th>
      <td class="field-body">
        <a class="reference external" href="https://www.percona.com/software/pmm/quickstart">Installing Percona Monitoring and Management</a></td>
    </tr>
  </tbody>
</table>

Percona Monitoring and Management (PMM) is a free and open-source platform for managing and monitoring MySQL, MongoDB, and PostgreSQL performance.

## Release Highlights

> - Custom Certificates support for MySQL monitoring, including remote monitoring, will let you monitor MySQL instances with such security restrictions. This functionality might help monitor Google Cloud SQL for MySQL instances easier.

Author: ```suggestion
- Custom certificates help define proper security levels for remotely monitored MySQL instances, including Google Cloud SQL.
```

> - External monitoring cleanups on UI to make it easier to use. You can fill all parameters or provide the URL of your endpoint, and PMM will detect all needed parts.

Author: ```suggestion
- Usability improvements to the External Monitoring UI. When filling parameters, you can enter the parts of an endpoint (scheme, host, path) or let PMM automatically extract them from a URL.
```

- pg_stat_monitor 0.9.0 support. This change will give you compatibility with the latest version. More features coming later.

- Single-line install of PMM Server on supported Linux distros (this feature is in Technical Preview).

- DBaaS Changes: (this feature is in Technical Preview)

>     - Make it easier to preview DBaaS functionality, so users can quickly turn it ON on the Advanced settings page. ([Read more](../how-to/configure.md#advanced-settings))

Reviewer 2: ```suggestion
    - Make it easier to tinker with DBaaS functionality; users can quickly turn it ON/OFF on the Advanced settings page. ([Read more](../how-to/configure.md#advanced-settings))
```

Author: 'tinker with' is what you do to a pre-war Austin. Maybe 'adjust'

Author: 'experience'

>     - Database components management will enable PMM administrators to restrict some DB versions from creating DB Clusters.

Reviewer 2: ```suggestion
    - Database components management will enable PMM administrators to limit users in your organization to specific (admin approved) database versions in their DBaaS DB Clusters.
```

>     - For PXC clusters created using DBaaS, the HAProxy will be used by default.  Please note: the proper monitoring HAProxy with DBaaS will come later.

Reviewer 2: ```suggestion
    - For PXC clusters created using DBaaS, HAProxy will now be used by default.  Please note: Monitoring of the HAProxy in DBaaS will be enabled in an upcoming release. 
```

- Changes to *Sign in to Percona Platform*. From this release, Registration of the Percona account will be more secure and require additional confirmation.

## New Features

* [PMM-7863](https://jira.percona.com/browse/PMM-7863): DBaaS: Ability to specify in K8s configuration the version of HAProxy to be used for DB creation 
* [PMM-7848](https://jira.percona.com/browse/PMM-7848), [PMM-7847](https://jira.percona.com/browse/PMM-7847), [PMM-7421](https://jira.percona.com/browse/PMM-7421): Add support for using SSL certificates between pmm-admin and monitored MySQL databases

> * [PMM-7883](https://jira.percona.com/browse/PMM-7883): Single-line install of PMM Server on supported Linux distros - [Technical Preview]

Reviewer 2: Nothing to fix here but we have the "quickstart" which has LOTS of steps...marketing owns that I think...need to update that to reference this as new pmm-server install steps.

* [PMM-7013](https://jira.percona.com/browse/PMM-7013), [PMM-7819](https://jira.percona.com/browse/PMM-7819): DBaaS: Use HAProxy by default instead of ProxySQL for MySQL DB clusters
* [PMM-7356](https://jira.percona.com/browse/PMM-7356), [PMM-7581](https://jira.percona.com/browse/PMM-7581): DBaaS: Management of available versions of DB components
* [PMM-7358](https://jira.percona.com/browse/PMM-7358), [PMM-7576](https://jira.percona.com/browse/PMM-7576): DBaaS: Management of default versions of DB components

## Improvements

* [PMM-7572](https://jira.percona.com/browse/PMM-7572): Add TLS options to mysqld_exporter
* [PMM-7783](https://jira.percona.com/browse/PMM-7783): Support of pg_stat_monitor v0.9.0
* [PMM-7064](https://jira.percona.com/browse/PMM-7064): Integrated Alerting: Presenting severity of the Alert Rule using different colors
* [PMM-7946](https://jira.percona.com/browse/PMM-7946): Better error message on PMM client if server doesn't support HAProxy
* [PMM-7932](https://jira.percona.com/browse/PMM-7932): Usability improvements on UI for adding External Services

> * [PMM-7641](https://jira.percona.com/browse/PMM-7641), [PMM-7820](https://jira.percona.com/browse/PMM-7820): Make DBAAS feature flag - same as other Feature Flags

Reviewer 2: ```suggestion
* [PMM-7641](https://jira.percona.com/browse/PMM-7641), [PMM-7820](https://jira.percona.com/browse/PMM-7820): Add DBaaS to Technical Preview section and allow user to Enable/Disable via UI
```

## Bugs Fixed

* [PMM-7911](https://jira.percona.com/browse/PMM-7911): DBaaS: Invalid Number of Nodes results in bad request
* [PMM-7884](https://jira.percona.com/browse/PMM-7884): DBaaS: Fix DB Cluster tab loading

> * [PMM-7941](https://jira.percona.com/browse/PMM-7941): mongodb_exporter provides a wrong replication status value

Reviewer 3: actually this one didn't get into the release. We found out just recently that it is not included.

Author: PMM-7941 was removed
